Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs

```diff
@@ -0,0 +1,82 @@
+use num_bigint::{BigUint, ParseBigIntError};
+use std::io;
+
+const DAYS_IN_YEAR: u64 = 365;
+
+// convert input string to BigUint
+fn read_parse(inp: &str) -> Result<BigUint, ParseBigIntError> {
+    let age = inp.trim().parse::<BigUint>();
+    age
+}
+
+// turn age in years to days
+fn calc_days(age: &BigUint) -> BigUint {
+    age * DAYS_IN_YEAR
+}
+
+fn main() {
+    println!("Enter age:");
+
+    let mut input = String::new();
+
+    io::stdin()
+        .read_line(&mut input)
+        .expect("Failed to read line.");
+
+    let age: BigUint = read_parse(&input).expect("Not a positive integer.");
+    println!("You entered: {}.", age);
+    println!("That's roughly {} days.", calc_days(&age));
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn an_ordinary_human_age() {
+        let result = read_parse(&String::from("50"));
+        assert_eq!(result, Ok(BigUint::from(50_u32)));
+    }
+
+    #[test]
+    fn age_of_earth() {
+        let result = read_parse(&String::from("4500000000"));
+        assert_eq!(result, Ok(BigUint::from(4500000000_u64)));
+    }
+
+    #[test]
+    #[should_panic(expected = "ParseBigIntError")]
+    fn not_a_positive_integer() {
+        read_parse(&String::from("boo")).unwrap();
+    }
+
+    #[test]
+    fn calc_days_for_earth() {
+        let result = calc_days(&read_parse(&String::from("4500000000")).unwrap());
```

Comment:
```suggestion
        let result = calc_days(BigUint::from(4500000000_u64));
```
All the tests for `calc_days` also use `read_parse`, which may be testing too many things at once. We want to test the target function in isolation as much as possible. Also for the other tests.